frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -381,6 +384,22 @@ public void shutdown() {
             model.removeJobQueue(backendChannel.id().asLongText());
             backendChannel.close();
         }
+        if (this.serverThread && this.connector != null) {
+            logger.debug("Cleaning connector socket");
+            this.connector.clean();
+        }
+        logger.debug("Terminating IOStreams for worker thread shutdown");
+        lifeCycle.terminateIOStreams();
+        try {
+            if (out != null) {
```

Comment:
When would we have out or error equal to null? Shoudln't they always be created?